src/coreclr/jit/codegenxarch.cpp

```diff
@@ -4397,6 +4398,37 @@ void CodeGen::genCodeForShift(GenTree* tree)
             inst_RV_SH(ins, size, tree->GetRegNum(), shiftByValue);
         }
     }
+#if defined(TARGET_64BIT)
+    else if (compiler->compOpportunisticallyDependsOn(InstructionSet_BMI2) && tree->OperIs(GT_LSH, GT_RSH, GT_RSZ) &&
+             (genActualType(targetType) == TYP_LONG))
```

Comment:
Why only for `TYP_LONG` and not for 32-bit shifts?

Reply:
And we could generate that on x86 as well I guess.

Reply:
Removed the condition and updated LSRA to enable 32 bit shifts.

Reply:
> And we could generate that on x86 as well I guess.

Test passed for x86 as well. Changing the title of this PR.

Reply:
Double checked that it was not enabled for x86. I will not handle x86 optimization in this PR and open an issue as a future work item.

Reply:
Double checked that it was not enabled for x86. I will not handle x86 optimization in this PR and open an issue as a future work item.